driver/bpf/Makefile

```
@@ -16,6 +16,24 @@ KERNELDIR ?= /lib/modules/$(shell uname -r)/build
 
 # DEBUG = -DBPF_DEBUG
 
+#
+# https://chromium.googlesource.com/chromiumos/third_party/kernel/+/096925a44076ba5c52faa84d255a847130ff341e%5E%21/#F2
+# This commit diverged the ChromiumOS kernel from stock in the area of audit information, which this probe accesses.
+# 
+# This enables the workaround for this divergence.
+#
+NEEDS_COS_73_WORKAROUND = $(shell expr `grep -sq "^\s*struct\s\+audit_task_info\s\+\*audit;\s*$$" $(KERNELDIR)/include/linux/sched.h` = 0)
```

Trying it locally I obtain something like this:

```
❯ make bpf

expr: syntax error
expr: syntax error
scripts/Makefile.lib:8: 'always' is deprecated. Please use 'always-y' instead
expr: syntax error
Built target bpf
```
To avoid this I would use grep with `-c` option:

```suggestion
NEEDS_COS_73_WORKAROUND = $(shell expr `grep -sc "^\s*struct\s\+audit_task_info\s\+\*audit;\s*$$" $(KERNELDIR)/include/linux/sched.h` = 1)
```
We just count the number of lines and the logic should work without errors since `grep` returns always a number unless the `$KERNELDIR` is not there, but in this case, the error message seems reasonable to me. Let me know if this sounds good to you

---

Thanks @Andreagit97 - I'm not sure I understand the error or the proposed fix, though. Does `expr` not work on program return codes in your shell, so using `-c` allows it to process stdout? Happy to use your fix either way, just wanted to fully understand it :)

---

Hi @Stringy, yes this is exactly the problem. `expr` cannot catch the return value of grep, but with `-c` option grep always returns the number of lines matching the pattern. In this way, `expr` can work with the stdout produced by grep.